leapp/reporting/__init__.py

```diff
@@ -125,6 +126,23 @@ def __init__(self, value=None):
         self._value = value
 
 
+class Key(BasePrimitive):
+    """Stable identifier for report entries"""
+    name = 'key'
+
+    def __init__(self, entries):
+
+        def _find_entry(cls):
+            entry = next((e for e in entries if isinstance(e, cls)), None)
+            if entry:
+                return entry._value
```

Comment:
No principal comment:
seems it is not needed to be inside the constructor

Reply:
But it's a helper function that is not needed anywhere else anyway.

Reply:
Outside of the constructor. Now each time you instantiate the Key a new function object of _find_entry is created. If you would have the function in the outer scope, then only one object will be presented. There are no reasons to keep it as a nested function, basically. But as the result of this solution, you will have some downsides, i.e. hard to mock, hard to unit test.

As an alternative you can make it as:
```
@classmethod
def _find_entry(cls):
    """Docs."""
```

Reply:
Problem with mocking and with unit testing linked with the problem to access the _find_entry object, because it is not available in the outer scope.  

Unit tests are exactly for testing such things like your function (_find_entry performs a piece of distinct functionality). Integration tests are for testing the whole thing. Unit test means testing each single unit, which allows you to focus on testing the very tiny piece of you functionality - a function and think about corner cases within a smaller scope.

Anyway, as you see in the header of my comment - it is not principle, so the only reason to maintain this thread is it is useful for me and you and it is not blocking it from progress. So consider resolving this treah anytime you want.